GuessSTARTTLSPolicies.py

```diff
@@ -0,0 +1,681 @@
+#!/usr/bin/python
+import sys
+import os
+import smtplib
+import subprocess
+import re
+import json
+import argparse
+import hashlib
+import dns.resolver
+import time
+import datetime
+import platform
+import copy
+from M2Crypto import X509
+
+from Config import Config
+from Errors import *
+from Utils import mkdirp, tls_protocols_higher_than, extract_pem_data
+
+CACHE_TIME = 86400  #seconds, 86400 = 1 day
+CACHE_DIR = ""
+
+def extract_names(x509_cert):
+  """Return a set of DNS subject names from PEM-encoded leaf cert."""
+
+  subj = x509_cert.get_subject()
+  # Certs have a "subject" identified by a Distingushed Name (DN).
+  # Host certs should also have a Common Name (CN) with a DNS name.
+  common_names = subj.get_entries_by_nid(subj.nid["CN"])
+  common_names = [name.get_data().as_text().lower() for name in common_names]
+  try:
+    # The SAN extension allows one cert to cover multiple domains
+    # and permits DNS wildcards.
+    # http://www.digicert.com/subject-alternative-name.htm
+    # The field is a comma delimited list, e.g.:
+    # >>> twitter_cert.get_ext('subjectAltName').get_value()
+    # 'DNS:www.twitter.com, DNS:twitter.com'
+    alt_names = x509_cert.get_ext("subjectAltName").get_value()
+    alt_names = alt_names.split(",")
+    alt_names = [name.strip().partition(":") for name in alt_names]
+    alt_names = [name.lower() for prot, _, name in alt_names if prot == "DNS"]
+  except:
+    alt_names = []
+  return set(common_names + alt_names)
+
+def get_certs_info(openssl_output):
```

Comment:
While we should migrate to the other openssl library, this is the main function I was thinking of lifting logic from.